pkg/distributor/distributor.go

```diff
@@ -350,7 +351,14 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	defer cancel() // cancel the timeout to release resources
 	for ingester, samples := range samplesByIngester {
 		go func(ingester *ring.IngesterDesc, samples []*sampleTracker) {
-			d.sendSamples(ctx, ingester, samples, &pushTracker)
+			// Use a background context to make sure all ingesters get samples even if we return early
+			localCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Do we still need the `WithTimeout` on line 350?  
Could we use the same timeout parameter?

Reply:
I like removing WithTimeout on line 350 and using the same timeout parameter for the background calls.

I believe if we don't have the background context, the context from our request is cancelled as soon as the response is returned.